js/binancetr.js

```diff
@@ -0,0 +1,308 @@
+        //                ["46202.47000000","0.00170000"],
+        //                ["46202.46000000","0.01378000"],
+        //                ["46202.30000000","0.01634000"]
+        //            ],
+        //            "asks":[
+        //                ["46203.22000000","0.01611000"],
+        //                ["46203.24000000","0.00498000"],
+        //                ["46203.26000000","0.01136000"],
+        //                ["46205.59000000","0.00216000"],
+        //                ["46205.84000000","0.08579000"]
+        //            ]
+        //        },
+        //        "timestamp":1641383770656
+        //    }
+        //
+        const result = this.safeValue (response, 'data', {});
+        const timestamp = this.safeInteger (response, 'timestamp', {});
+        const orderBook = this.parseOrderBook (result, symbol, timestamp, 'bids', 'asks', 0, 1);
+        orderBook['nonce'] = this.safeInteger (result, 'lastUpdateId', {});
+        return orderBook;
+    }
+
+    async fetchTrades (symbol, since = undefined, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const request = {
+            'symbol': market['id'],
+        };
+        if (limit !== undefined) {
+            request['limit'] = limit; // default: 500
+        }
+        const response = await this.publicGetTrades (this.extend (request, params));
+        //
+        //  [
+        //    {"id":1211494273,"price":"46410.42000000","qty":"0.00084000","quoteQty":"38.98475280","time":1641391959076,"isBuyerMaker":false,"isBestMatch":true},
+        //    {"id":1211494274,"price":"46410.42000000","qty":"0.00084000","quoteQty":"38.98475280","time":1641391959278,"isBuyerMaker":false,"isBestMatch":true},
+        //    {"id":1211494275,"price":"46410.41000000","qty":"0.00061000","quoteQty":"28.31035010","time":1641391959289,"isBuyerMaker":true,"isBestMatch":true},
+        //    {"id":1211494276,"price":"46410.42000000","qty":"0.00084000","quoteQty":"38.98475280","time":1641391959357,"isBuyerMaker":false,"isBestMatch":true},
+        //    {"id":1211494277,"price":"46410.42000000","qty":"0.00084000","quoteQty":"38.98475280","time":1641391959483,"isBuyerMaker":false,"isBestMatch":true},
+        //  ]
+        //
+        const data = this.safeValue (response, []);
+        return this.parseTrades (data, market, since, limit);
+    }
+
+    sign (path, api = 'public', method = 'GET', params = {}, headers = undefined, body = undefined) {
+        let url = this.urls['api'][api] + this.version + '/' + path;
+        if (api === 'public') {
+            if (Object.keys (params).length) {
+                url += '?' + this.urlencode (params);
+            }
+        }
+        if (api === 'public' && path === 'trades') {
+            url = this.urls['api']['public3'] + path;
+            params['symbol'] = params['symbol'].replace ('_', '');
```

Comment:
We should not parse the `symbol` param in the `sign()` method – is there a reason to do it here?

Reply:
In binancetr the symbol needs to be BTC_USDT, but in the URL it needs to be sent as BTC_USDT for some functions and  BTCUSDT for others. Thats why I modify only when the URL is generated for that specific function.

Reply:
Yes, but i don't see any v3 calls throughout the file. Do we actually need it? It looks like you're not handling "type-2 "symbols anywhere?

The problem is – just removing the underscore from there upon sending the request – won't work, cause we also have to recognize those symbols upon parsing them.

1. Let's remove this line from `sign()`. 
2. For methods that use a v3-endpoint without underscore `_`, let's define the symbol like this upon sending the request:
```
    async fetchOrderBook (symbol, limit = undefined, params = {}) {
        await this.loadMarkets ();
        const market = this.market (symbol);
        const request = {
            'symbol': market['base'] + market['quote'],
        };
```

Let's also make sure that all symbols work with fetchOrderBook as expected.

Reply:
Ok, changed.